third_party/istio-1.0.6/download-istio.sh

```diff
@@ -1,10 +1,14 @@
+#!/usr/bin/env bash
+
 # Download and unpack Istio
 ISTIO_VERSION=1.0.6
 DOWNLOAD_URL=https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-linux.tar.gz
 
 wget $DOWNLOAD_URL
 tar xzf istio-${ISTIO_VERSION}-linux.tar.gz
-cd istio-${ISTIO_VERSION}
+
+( # subshell in downloaded directory
+cd istio-${ISTIO_VERSION} || exit
```

Comment:
I think this will just exit the subshell - did you want it to exit the whole thing?
you may want to give it a non-zero exit code either way

Reply:
also, show we add "set -e" to these so that things stop on failure? then we don't need to special case the 'cd'

Reply:
Just exiting the subshell is fine, the cleanup can happily run afterwards.